src/Bicep.Core/TypeSystem/Providers/IProviderDescriptor.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Registry;
+
+namespace Bicep.Core.TypeSystem.Providers;
+
+public interface IProviderDescriptor
```

Comment:
There are no references to this interface or the `LegacyProviderDescriptor` class. Can you remove them?